scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -199,6 +199,24 @@ sealed trait Execution[+T] extends java.io.Serializable { self: Product =>
   def zip[U](that: Execution[U]): Execution[(T, U)] =
     Zipped(this, that)
 
+  /**
+   * Add an extra config for current execution.
+   * This changes of config are isolated and affect only current execution.
+   * All changes will disappear for any next operation on execution.
+   */
+  def addConfig(extraConfig: Config): Execution[T] =
```

Comment:
I like this style better than the static `.withConfig`, but I think it will only serve to clutter the API. If you want this style, my preference would be to have a method of the same name or perhaps `#transformConfig` on `trait Execution` that makes it trivial to do this:

```scala
val extraConfig: Config = /* ... */

def myExecution(exec: Execution[Unit]): Execution[Unit] = {
  exec
    .transformConfig(_ ++ extraConfig)
    .flatMap { /* ... */ }
    // ...
}
```

and maybe deprecate the `.withConfig` method.